src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,162 @@
+                        } catch (TodoException e) {
+                            System.err.println(e.getMessage());
+                        }
+                        break;
+
+                    case ("deadline"):
+                        System.out.println(line);
+                        tempArr = (command.substring(9)).split("/");
+                        Deadline currDeadline = new Deadline(tempArr[0], tempArr[1].substring(3));
+                        myList.add(currDeadline);
+                        System.out.println("Got it. I've added this task:");
+                        System.out.println(currDeadline);
+                        System.out.println("Now you have " + myList.size() + " tasks in the list.");
+                        System.out.println(line);
+                        storageFile.updateStorageFile(myList);
+                        break;
+
+                    case ("event"):
+                        System.out.println(line);
+                        tempArr = (command.substring(6)).split("/");
+                        Event currEvent = new Event(tempArr[0], tempArr[1].substring(5), tempArr[2].substring(3));
+                        myList.add(currEvent);
+                        System.out.println("Got it. I've added this task:");
+                        System.out.println(currEvent);
+                        System.out.println("Now you have " + myList.size() + " tasks in the list.");
+                        System.out.println(line);
+                        storageFile.updateStorageFile(myList);
+                        break;
+                }
+                command = scanner.nextLine(); // Read next command
+                continue;
+            }
+
+            if (command.contains("mark")) {
+                tempArr = command.split(" ");
+                currTask = myList.get(Integer.parseInt(tempArr[1]) - 1);
+                switch (tempArr[0]) {
+                    case ("mark"):
+                        System.out.println(line);
+                        currTask.setDone();
+                        System.out.println("Nice! I've marked this task as done:");
+                        System.out.println(currTask);
+                        System.out.println(line);
+                        storageFile.updateStorageFile(myList);
+                        break;
+
+                    case ("unmark"):
+                        System.out.println(line);
+                        currTask.setNotDone();
+                        System.out.println("Ok, I've marked this task as not done yet:");
+                        System.out.println(currTask);
+                        System.out.println(line);
+                        storageFile.updateStorageFile(myList);
+                        break;
+
+                    default:
+                        System.out.println("Wrong Command");
+                }
+                command = scanner.nextLine(); // Read next command
+                continue;
+            }
```

Comment:
Perhaps we can refactor the command interpretation into the Command class as a static method?